internal/ui/src/components/id.tsx

```diff
@@ -0,0 +1,79 @@
+"use client";
+import { Slot } from "@radix-ui/react-slot";
+import * as React from "react";
+import { cn } from "../lib/utils";
+import { useState } from "react";
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "./ui/tooltip";
+import { CopyIcon } from "lucide-react";
+
+
+
+type copyProps = React.SVGProps<SVGSVGElement> & {
+    width?: number;
+    height?: number;
+};
+
+
+type IdProps = {
+    value?: string;
+    truncate?: number;
+    className?: string;
+};
+
+export const Id: React.FC<IdProps> = ({ className, value, truncate, ...props }) => {
+    const [isHover, setIsHover] = useState(false);
+    const [iconHover, setIconHover] = useState(false);
+    const copyTextToClipboard = async (value: string) => {
+        try {
+            await navigator.clipboard.writeText(value ?? "");
+        } catch (error) {
+            console.error('Failed to copy: ', error);
+        }
+    };
+
+    function generateString() {
+        const char = '•';
+        let length = 0;
+
+        if (!value?.length || !truncate) {
+            return '';
+        } else {
+            length = value?.length - truncate;
+        }
+        return Array.from({ length }, () => char).join('');
+    }
+
+    const truncateValue = truncate ? value?.slice(0, truncate) + generateString() : value;
+    const Comp = "div";
+
+    return (
+        <Comp
+            className={cn("inline-flex group items-center transition duration-150 justify-center gap-3 whitespace-nowrap tracking-normal rounded-lg  font-medium transition-colors disabled:pointer-events-none focus:outline-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-base-12 w-fit max-w-96 border border-accent-6 text-accent-12 radius radius-2 font-mono font-normal h-8 px-3 py-1 text-xs", className)}
+            onClick={() => copyTextToClipboard(value ?? "Copied")}
+            {...props}
+            onMouseEnter={() => setIsHover(true)}
+            onFocus={() => setIsHover(true)}
+            onMouseLeave={() => setIsHover(false)}
+            onBlur={() => setIsHover(false)}
+        >
+            {truncateValue}
+            {isHover &&
+                <TooltipProvider>
```

Comment:
I'm not sure if we want one tooltip provider per field, could you check their docs and see what the recommended approach is?
1 global provider vs many smaller ones

Reply:
![CleanShot 2024-12-03 at 10 16 16@2x](https://github.com/user-attachments/assets/e325bde0-59aa-4644-bea9-8a715e513e83)

Reply:
This was the plan just had not gotten there yet. Fixed.